### PR TITLE
added index, show, create, update, destroy methods to channels_contro…

### PR DESCRIPTION
**Pull request by lali20021**
…ller.rb

### CODE REVIEW

File: app/controllers/channels_controller.rb [hunk 1]
```diff
@@ -1,2 +1,45 @@
...
+            render json: {error: "Channel wasn't updated"}, status: 400
+        end 
+    end 
+    
+    def destroy
+        @channel = current_user.channels.find(params[:id])
+        if @channel.destroy 
+        render json: @channel
+        else 
+            render json: {error: "Could't delete"}, status: 400
```
Comment (Dimitriedd) [inline app/controllers/channels_controller.rb]: Couldn't*

**Comment by winescout** [comment]
Looks good

File: app/controllers/channels_controller.rb [hunk 2]
```diff
@@ -1,2 +1,45 @@
...
+        @channel = current_user.channels.create(channel_params)
+        if @channel.save 
+            render json: @channel, status: 201
+        else 
+            render json: {error: "Channel wasn't saved"}, status: 400
+        end 
+    end 
+    
+    def update 
+        @channel = current_user.channles.find(params[:id])
```
Comment (joshalexgibs) [inline app/controllers/channels_controller.rb]: Typo here: "current_user.channles... => current_user.channels..."

File: app/controllers/channels_controller.rb [hunk 3]
```diff
@@ -1,2 +1,45 @@
 class ChannelsController < ApplicationController
+    before_action :authenticate_user!, :except => [:index]
+    
+    def index 
+        @channels = Channel.all 
+        render json: @channels
+    end 
+    
+    def show 
+        @channel = Channel.find(params[:id])
```
Comment (joshalexgibs) [inline app/controllers/channels_controller.rb]: missing "render json: @channel"

File: app/controllers/channels_controller.rb [hunk 4]
```diff
@@ -1,2 +1,45 @@
...
+            render json: @channel
+        else 
+            render json: {error: "Channel wasn't updated"}, status: 400
+        end 
+    end 
+    
+    def destroy
+        @channel = current_user.channels.find(params[:id])
+        if @channel.destroy 
+        render json: @channel
```
Comment (joshalexgibs) [inline app/controllers/channels_controller.rb]: Missing indent, nothing code-breaking.

File: app/controllers/channels_controller.rb [hunk 5]
```diff
@@ -1,2 +1,45 @@
...
+            render json: @channel, status: 201
+        else 
+            render json: {error: "Channel wasn't saved"}, status: 400
+        end 
+    end 
+    
+    def update 
+        @channel = current_user.channles.find(params[:id])
+        if @channel.update(channel_params)
+            render json: @channel
```
Comment (joshalexgibs) [inline app/controllers/channels_controller.rb]: missing "status: 201"

**Pull request merged**